File: ruby-gem/lib/models/shipping_option.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #ShippingOption
require_relative '../core/semantic_object'
require_relative 'how_subject'


module DfcLinkmlConnector
  module Models
    class ShippingOption < HowSubject
      SEMANTIC_TYPE = "dfc-b:ShippingOption".freeze

      # @return [String]
      attr_accessor :end_date

      # @return [Float]
      attr_accessor :fee

      # @return [Float]
      attr_accessor :quantity

      # @return [String]
      attr_accessor :start_date

      # @return [String]
      attr_accessor :option_of

      # @return [String]
      attr_accessor :selected_by

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, endDate: nil, fee: nil, quantity: nil, startDate: nil, optionOf: nil, selectedBy: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, endDate: nil, fee: nil, quantity: nil, startDate: nil, optionOf: nil, selectedBy: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @end_date = endDate
        @fee = fee
        @quantity = quantity
        @start_date = startDate
        @option_of = optionOf
        @selected_by = selectedBy
        self.semanticType = "dfc-b:ShippingOption"
        registerSemanticProperty("dfc-b:endDate", &method("end_date")).valueSetter = method("end_date=")
        registerSemanticProperty("dfc-b:fee", &method("fee")).valueSetter = method("fee=")
        registerSemanticProperty("dfc-b:quantity", &method("quantity")).valueSetter = method("quantity=")
        registerSemanticProperty("dfc-b:startDate", &method("start_date")).valueSetter = method("start_date=")
        registerSemanticProperty("dfc-b:optionOf", &method("option_of")).valueSetter = method("option_of=")
        registerSemanticProperty("dfc-b:selectedBy", &method("selected_by")).valueSetter = method("selected_by=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end

File: typescript-connector/src/models/ShippingOption.ts
// Class from DFC Business Ontology: #ShippingOption
import { SemanticObject } from "../core/SemanticObject.js";
import { HowSubject, type HowSubjectParams } from "./HowSubject.js";
import type { QuantitativeValue } from "./QuantitativeValue.js";
import type { SaleSession } from "./SaleSession.js";

export interface ShippingOptionParams extends HowSubjectParams {
  endDate?: string;
  fee?: number;
  quantity?: number;
  startDate?: string;
  selectedBy?: string;
  hasQuantity?: QuantitativeValue | string;
  optionOf?: SaleSession | string;
}
export class ShippingOption extends HowSubject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:ShippingOption";
  }

  endDate?: string;
  fee?: number;
  quantity?: number;
  startDate?: string;
  selectedBy?: string;
  hasQuantity?: QuantitativeValue | string;
  optionOf?: SaleSession | string;

  constructor(
    semanticId: string,
    params?: ShippingOptionParams,
  ) {
    super(semanticId, params);
    this.endDate = params?.endDate;
    this.fee = params?.fee;
    this.quantity = params?.quantity;
    this.startDate = params?.startDate;
    this.selectedBy = params?.selectedBy;
    this.hasQuantity = params?.hasQuantity;
    this.optionOf = params?.optionOf;
    this.semanticType = ShippingOption.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:endDate", () => this.endDate);
    this.registerSemanticProperty("dfc-b:fee", () => this.fee);
    this.registerSemanticProperty("dfc-b:quantity", () => this.quantity);
    this.registerSemanticProperty("dfc-b:startDate", () => this.startDate);
    this.registerSemanticProperty("dfc-b:selectedBy", () => this.selectedBy);
    this.registerSemanticProperty("dfc-b:hasQuantity", () => this.hasQuantity);
    this.registerSemanticProperty("dfc-b:optionOf", () => this.optionOf);
  }
  static {
    SemanticObject.typeRegistry.set(ShippingOption.SEMANTIC_TYPE, ShippingOption);
  }
}
